scenarios/aws/ec2/vm_run.go

```diff
@@ -75,16 +75,16 @@ func VMRunWithDocker(ctx *pulumi.Context) error {
 
 	// If no OS is provided, we default to AmazonLinuxECS as it ships with Docker pre-installed
 	osDesc := os.DescriptorFromString(env.InfraOSDescriptor(), os.AmazonLinuxECS)
-	vm, err := NewVM(env, "vm", WithAMI(env.InfraOSImageID(), osDesc, osDesc.Architecture))
+	vm, err := NewVM(env, "vm", WithAMI(env.InfraOSImageID(), osDesc, osDesc.Architecture), WithInstanceProfile(env.DefaultDockerInstanceProfile()))
```

Comment:
It should be defaulted in `resources/aws/ec2/vm.go` instead as I don't think we need to only use it for Docker case.